New rust plugin fails to validate with a `rust-deps` part dependency

### Bug Description

We allow a `rust-deps` part dependency in the rust plugin so the user may install rust in the manner they choose.

The updated rust plugin, however, fails to validate when this occurs.

### To Reproduce

Modify the `test_get_pull_commands_compat` test in `test_rust_plugin.py` as follows:

```python
def test_get_pull_commands_compat(part_info):
    properties = RustPlugin.properties_class.unmarshal(
        {"source": ".", "after": ["rust-deps"]}
    )
    plugin = RustPlugin(properties=properties, part_info=part_info)
    plugin.validator_class(
        part_name="project",
        env="",
        properties=properties
    ).validate_environment(part_dependencies=["rust-deps"])
    plugin._check_rustup = lambda: False

    commands = plugin.get_build_commands()
    assert plugin.get_pull_commands() == []
    assert 'cargo install -f --locked --path "."' in commands[0]
```

### part yaml

```shell
parts:
  rust-deps:
    plugin: nil
  project:
    source: .
    after: ["rust-deps"]
```


### Relevant log output

```shell
FAILED   [100%]
tests/unit/plugins/test_rust_plugin.py:184 (test_get_pull_commands_compat)
part_info = <craft_parts.infos.PartInfo object at 0x7fdbcf9c8850>

    def test_get_pull_commands_compat(part_info):
        properties = RustPlugin.properties_class.unmarshal(
            {"source": ".", "after": ["rust-deps"]}
        )
        plugin = RustPlugin(properties=properties, part_info=part_info)
>       plugin.validator_class(
            part_name="project",
            env="",
            properties=properties
        ).validate_environment(part_dependencies=["rust-deps"])

/home/lengau/Work/Code/craft-parts/tests/unit/plugins/test_rust_plugin.py:190: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/home/lengau/Work/Code/craft-parts/craft_parts/plugins/rust_plugin.py:103: in validate_environment
    options.rust_channel = "none"  # type: ignore[misc]
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   ???
E   TypeError: "RustPluginProperties" is immutable and does not support item assignment

pydantic/main.py:359: TypeError
```

### Other information
When fixing this bug, remove the type  linter ignore comment in `rust_plugin.py` that mentions this bug.

--- a/craft_parts/plugins/rust_plugin.py
+++ b/craft_parts/plugins/rust_plugin.py
@@ -239,8 +239,6 @@ sh -s -- -y --no-modify-path --profile=minimal --default-toolchain {channel}
         for crate in options.rust_path:
             logger.info("Generating build commands for %s", crate)
             config_cmd_string = " ".join(config_cmd)
-            # TODO(liushuyu): the current fallback installation method will also install
-            # compiler plugins into the final Snap, which is likely not what we want.
             # pylint: disable=line-too-long
             rust_build_cmd_single = dedent(
                 f"""\
@@ -254,9 +252,12 @@ sh -s -- -y --no-modify-path --profile=minimal --default-toolchain {channel}
                     pushd "{crate}"
                     cargo build --workspace --release {config_cmd_string}
                     # install the final binaries
-                    # TODO(liushuyu): this will also install proc macros in the workspace,
-                    # which the user may not want to keep in the final installation
                     find ./target/release -maxdepth 1 -executable -exec install -Dvm755 {{}} "{self._part_info.part_install_dir}" ';'
+                    # remove proc_macro objects
+                    for i in "{self._part_info.part_install_dir}"/*.so; do
+                        readelf --wide --dyn-syms "$i" | grep -q '__rustc_proc_macro_decls_[0-9a-f]*__' && \
+                        rm -fv "$i"
+                    done
                     popd
                 fi\
                 """

--- a/craft_parts/plugins/rust_plugin.py
+++ b/craft_parts/plugins/rust_plugin.py
@@ -87,7 +87,7 @@ class RustPluginEnvironmentValidator(validator.PluginEnvironmentValidator):
         """
         if "rust-deps" in (part_dependencies or []):
             options = cast(RustPluginProperties, self._options)
-            if options.rust_channel:
+            if options.rust_channel and options.rust_channel != "none":
                 raise validator.errors.PluginEnvironmentValidationError(
                     part_name=self._part_name,
                     reason="rust-deps can not be used"
@@ -99,12 +99,6 @@ class RustPluginEnvironmentValidator(validator.PluginEnvironmentValidator):
                     plugin_name="rust",
                     part_dependencies=part_dependencies,
                 )
-            # The following type ignore is to make type checking pass while
-            # issue #559 is a bug:
-            # https://github.com/canonical/craft-parts/issues/559
-            # The ignore comment (and this block comment) should be removed
-            # with the fixing of #559.
-            options.rust_channel = "none"  # type: ignore[misc]
 
 
 class RustPlugin(Plugin):

--- a/setup.py
+++ b/setup.py
@@ -88,7 +88,7 @@ test_requires = [
     "isort",
     "hypothesis",
     "pydocstyle",
-    "pylint",
+    "pylint<3.0",
     "pylint-fixme-info",
     "pylint-pytest",
     "pyright==1.1.327",

